# Term Project Progress Notebook - Manipulation and Analysis (NEW)

- Lindsey Rojtas
- LING1340
- ler75@pitt.edu

Welcome to the second Notebook! It should be noted that this is a **continuation** of `progress_data`! My progress report 3 starts there and will end here. 

## PROGRESS REPORT 3 CONTINUED

Before anything else, we should load in those pickle files. 

In [1]:
import pandas as pd

In [2]:
byfile = pd.read_pickle('../privdata/byfile_df.pkl')
bypar = pd.read_pickle('../privdata/byparticipant_df.pkl')
wordlist = pd.read_pickle('../privdata/wordlist_df.pkl')

In [3]:
byfile.head()

file              participants  \
0  data001.txt  [F107, F023, M023, F128]   
1  data002.txt        [F107, F023, F128]   
2  data003.txt              [F033, F056]   
3  data004.txt              [M018, F128]   
4  data005.txt  [M023, F128, F116, M026]   

                                             content  
0  [F107：＊＊＊の町というのはちいちゃくって、城壁がこう町全体をぐるっと回ってて、それが城...  
1  [F107：今度はーイギリスにもアメリカと同様のテロが起こるだろうって言ったんだってよ。, ...  
2  [F033：倒れちゃう。, F056：いきなり倒れた。, F033：どうしよう。あっ、この間...  
3  [F128：いや、別にいいよ。ローソンでいいやろ。ちょっと倒していい、これ。どうよ、調子は。...  
4  [F128：来てたときによく貸してもらったやつだ。, M023：そう、そんな感じのとこ。, ...

In [4]:
bypar.head()

participant        age  appears_count  \
0        F001  Early 20s              5   
1        F002  Late 60's              3   
2        F003  Late 80's              1   
3        F004  Late 20's             14   
4        F005  Late 20's              3   

                                          appears_in  \
0  [data105.txt, data086.txt, data076.txt, data07...   
1            [data033.txt, data032.txt, data031.txt]   
2                                      [data129.txt]   
3  [data096.txt, data094.txt, data092.txt, data08...   
4            [data052.txt, data023.txt, data015.txt]   

                                             content  
0  うーん、わかんない。そういうこと言わないで。うるさいな。うるさい。うるさいって言ってるの。う...  
1  ２７歳から現在まで東京都に居住。南仏へいらしたそうだけど、（うん）どうでしたか。みんな、太っ...  
2  そうねえ。＜笑い＞先生はね、師範卒業したのがねえ、１９歳だったのよ。だーから、若い先生でね。...  
3  うん、まあね。はい、もう始まってますからね。よろしくね。ちょっとちょっと、ちゃんとさ、つなぐ...  
4  はーい。いや、F034さんってー、やっぱりー、ハンバーガーとか好きですよねー。＊はなしを＊。...

In [5]:
wordlist.head()

katakana translation  frequency
0    パーセント     percent      63392
2     アメリカ     America      28243
3      ページ        page      24642
4     センター      center      20664
5     サービス     service      16630

### Exploratory Analysis and Developing Questions
So, let's get some structure in here. What I'm gonna do next is check the length of everything as well as check for some potential issues that will come about in our analysis. 

In [6]:
len(wordlist)

5192

So there are around 5,000 entries in our list of Katakana words.... 

In [7]:
len(bypar)

198

... 198 participants... 

In [8]:
len(byfile)

129

... who spoke in 129 files. It's pretty easy to see that not everyone spoke in the same amount of files for the same amount of time, so we should examine the data for potential outliers. One thing I can think of off the top of my head is how many times a certain participant was present in the conversations. We can see that F004 was present in 14 different files, so obviously she will have more conversation data than anyone else, unless her contributions in many of the files were marginal. Let's take a peek at some value counts:

In [9]:
bypar['appears_count'].value_counts()

1     162
2      16
3       7
5       4
4       4
7       2
14      1
11      1
6       1
Name: appears_count, dtype: int64

Ah, so we're seeing a problem here. The majority of the participants only appeared in one file, but there are quite a few outliers here. I could just exclude them from my analysis from the get-go, but we should probably see just how long the strings of text are for each participant. 

In [10]:
bypar['length'] = 0
for i in range(len(bypar)):
    bypar['length'][i] = len(bypar['content'][i])

C:\Users\lindsey rojtas\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [30]:
bypar.sample(5)

participant         age  appears_count     appears_in  \
67         F069   Early 20s              1  [data011.txt]   
98         F100           ?              1  [data103.txt]   
47         F049   Early 30s              1  [data053.txt]   
20         F022   Early 20s              1  [data027.txt]   
154        F156  Late teens              1  [data102.txt]   

                                               content  length  
67   密かに押してるでしょ。光ってるもん、ここ。＜笑い＞F149さんよりこういうのはわかってるわよ...   10316  
98                                                           0  
47   あ、それはねー、（うん）もともとー、（うん）とにかく物を作る方向には行きたかったのね。（うん...    6859  
20   ちゃんと一緒に住んで、（ふーん）あーでもちゃんと親との交流もあったみたいだし、（うーん）長か...    1842  
154  F156です。成人式の写真見ようよ。うん、だんだん。いやー、すげえかわいいじゃん。私、青。F...    8628

In [12]:
bypar['length'].describe()

count       198.000000
mean      11995.045455
std       14826.154061
min           0.000000
25%        5632.000000
50%        8186.000000
75%       12987.250000
max      133395.000000
Name: length, dtype: float64